book/from_flat_php_to_symfony2.rst

```diff
@@ -254,9 +254,11 @@ an individual blog result based on a given id::
     function get_post_by_id($id)
     {
         $link = open_database_connection();
-        $id = intval($id);
-        $result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);
-        $row = $result->fetch(PDO::FETCH_ASSOC);
+        $query = 'SELECT created_at, title, body FROM post WHERE  id=:id';
+        $statement = $pdo->prepare($query);
+        $statement->bindParam(':id', $id, PDO::PARAM_INT);
```

Comment:
yes you’re right, 
bindValue(...) makes more sense, since statement is not being re-used with different values in a bound param-variable

regards,

.. matt ..

> On 20 Jun 2016, at 13:21, Christian Flothmann <notifications@github.com> wrote:
> 
> In book/from_flat_php_to_symfony2.rst <https://github.com/symfony/symfony-docs/pull/6665#discussion_r67680106>:
> 
> > @@ -254,9 +254,11 @@ an individual blog result based on a given id::
> >      function get_post_by_id($id)
> >      {
> >          $link = open_database_connection();
> > -        $id = intval($id);
> > -        $result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);
> > -        $row = $result->fetch(PDO::FETCH_ASSOC);
> > +        $query = 'SELECT created_at, title, body FROM post WHERE  id=:id';
> > +        $statement = $pdo->prepare($query);
> > +        $statement->bindParam(':id', $id, PDO::PARAM_INT);
> bindValue()?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/symfony/symfony-docs/pull/6665/files/065dd3a3d49853eca691fb98003fb086a85b81c6#r67680106>, or mute the thread <https://github.com/notifications/unsubscribe/ABKRV9ecC9edXcKfJu2m8hGGpHyOEjl0ks5qNoXPgaJpZM4I5n_r>.
> 

--
Dr. Matt Smith | Senior Lecturer in Computing | matt.smith@itb.ie
Department of Informatics, School of Informatics & Engineering,
Institute of Technology Blanchardstown, Dublin 15, Republic of Ireland
www.itb.ie | Tel: +353-(1)-885-1098 | Fax: +353-(1)-885-1001

Recent publications: 
- Unity 5.x Cookbook (2015)